onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -946,12 +946,25 @@ def aten_batch_norm_update_stats(
     raise NotImplementedError()
 
 
-def aten_bernoulli(self: TensorType, generator: Optional[str] = None) -> TensorType:
+def aten_bernoulli(self: TensorType) -> TensorType:
     """bernoulli(Tensor self, *, Generator? generator=None) -> Tensor"""
 
     raise NotImplementedError()
 
 
+@torch_op("aten::bernoulli.p")
+def aten_bernoulli_p(self: TensorType, p: float) -> TensorType:
```

Comment:
Do we accept int inputs? If not, consider:

```suggestion
def aten_bernoulli_p(self: TFloat, p: float) -> TFloat:
```

Reply:
>>> op.p(torch.tensor([0, 1]), 0.5)
tensor([0, 0])

Let's keep it `TensorType`.

Reply:
Sorry could you explain? Does it mean the output type may be different than the input type?

Reply:
To be precise it should be the following
```
T1: {'tensor(int32)', 'tensor(uint32)', 'tensor(bool)', 'tensor(uint8)', 'tensor(uint16)', 'tensor(double)', 'tensor(int8)', 'tensor(float16)', 'tensor(string)', 'tensor(int64)', 'tensor(float)', 'tensor(int16)', 'tensor(uint64)'}
self: T1
return_val: T1
```
We probably don't have existing typing that precisely matches this so for me personally I'm fine with anything close ... `TTensor` looks good.

Reply:
Yeah that’s right. We are missing bfloat16 but I don’t think we need to care too much at this point. 

Alternatively if we do really care, we can cast self to some other type, since the op only needs the shape?

Reply:
Will do TTensor.